File: ABC227/C.cpp
#define _GLIBCXX_DEBUG //配列において[]で範囲外を参照したときにエラーを出してくれる
#include <bits/stdc++.h>
using namespace std;

int main(){
    long long N;
    cin >> N;
    long long A = 1;
    long long B = 1;
    long long C = 1;
    long long D = 0;
    long long ans = 0;
    while (A*A*A <= N){
        B = A;
        while (A*B*B <= N){
            ans += N/(A*B) - B + 1;
            B++;
        }
        A++;
    }
    cout << ans << endl;
}

File: ABC227/C.py
N = int(input())

A = 1
B = 1
C = 1
ans = 0
while A*A*A <= N:
    D = N//A
    B = A
    while B*B <= D :
        ans += D//B-B+1
        B += 1
    A += 1

print(ans)